cumulusci/cli/service.py

```diff
@@ -159,10 +159,21 @@ def callback(*args, **kwargs):
                 validator = import_global(validator_path)
                 validator(serv_conf)
 
+            ConfigClass = ServiceConfig
+            if "class_path" in service_config:
+                ConfigClass = import_class(service_config["class_path"])
+                if "connect" in dir(ConfigClass):
+                    client_info = runtime.keychain.get_service(
+                        "oauth-client", kwargs["oauth_client"]
+                    )
+                    oauth_dict = ConfigClass.connect(client_info)
```

Comment:
I think we should pass kwargs here and let the connect method look up the oauth_client. There may be services which need to get connected in a different way other than oauth.